engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java

```diff
@@ -67,22 +70,42 @@
     private ModuleEnvironment environment;
     private Context context;
     private WorldConfigurator oldWorldConfig;
+    private Name newWorldName;
+    private UIDropdownScrollable worldsDropdown;
 
     @Override
     public void initialise() {
         setAnimationSystem(MenuAnimationSystems.createDefaultSwipeAnimation());
 
         WidgetUtil.trySubscribe(this, "close", button -> {
+            final UniverseSetupScreen universeSetupScreen = getManager().createScreen(UniverseSetupScreen.ASSET_URI, UniverseSetupScreen.class);
+            final WorldPreGenerationScreen worldPreGenerationScreen =getManager().createScreen(WorldPreGenerationScreen.ASSET_URI, WorldPreGenerationScreen.class );
+            UIText customWorldName = find("customisedWorldName", UIText.class);
+            if(customWorldName.getText().isEmpty()==false) {
+                newWorldName = new Name(customWorldName.getText());
+                world.setWorldName(newWorldName);
+            }
+            universeSetupScreen.refresh(worldsDropdown);
+            worldPreGenerationScreen.setName(newWorldName);
             triggerBackAnimation();
         });
     }
 
+    /**
+     * This method sets the world name in title as well as in UITextBox
+     */
+    private void setName(UIText customWorldName) {
```

Comment:
Poor method name: if you are `set`ting the variable `customWorldName` the method should be called `setCustomWorldName` by convention.

Private methods don't _necessarily_ require javadoc, sometimes it can be useful if they're complex. Setters/getters usually don't justify it.